### PR TITLE
ConsularSites - Base structure

### PR DESCRIPTION
**Pull request by Tiriel**
- Add Consular role
- Add consular managed areas
- Add consular districts
- Add migration

### CODE REVIEW

File: src/Controller/EnMarche/EventManager/ConsularEventManagerController.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * @Route(path="/espace-consulaire", name="app_consular_event_manager_")
+ *
+ * @Security("is_granted('ROLE_CONSULAR')")
+ */
+class ConsularEventManagerController extends AbstractEventManagerController
+{
+    private $repository;
+
+    public function __construct(MunicipalEventRepository $repository)
```
Comment (ottaviano) [inline src/Controller/EnMarche/EventManager/ConsularEventManagerController.php]: `MunicipalEventRepository` => `ConsularEventRepository`?

File: src/Entity/ConsularManagedArea.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+     * @ORM\Column(type="integer")
+     * @ORM\Id
+     * @ORM\GeneratedValue
+     */
+    private $id;
+
+    /**
+     * @var ReferentTag[]|ArrayCollection
+     *
+     * @ORM\ManyToMany(targetEntity="AppBundle\Entity\ReferentTag")
```
Comment (ottaviano) [inline src/Entity/ConsularManagedArea.php]: 1) should be `@ManyToOne`, no? 
2) `targetEntity="AppBundle\Entity\ConsularDistrict"`

File: src/Entity/ConsularManagedArea.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+     * @var int
+     *
+     * @ORM\Column(type="integer")
+     * @ORM\Id
+     * @ORM\GeneratedValue
+     */
+    private $id;
+
+    /**
+     * @var ReferentTag[]|ArrayCollection
```
Comment (ottaviano) [inline src/Entity/ConsularManagedArea.php]: `@var ConsularDistrict` ?

File: src/Entity/ConsularManagedArea.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+     *
+     * @ORM\ManyToMany(targetEntity="AppBundle\Entity\ReferentTag")
+     */
+    private $districtTag;
```
Comment (ottaviano) [inline src/Entity/ConsularManagedArea.php]: `$consularDistrict`

File: src/Entity/ConsularDistrict.php [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+ * @ORM\Entity(repositoryClass="AppBundle\Repository\ConsularDistrictRepository")
+ * @ORM\Table(
+ *     name="consular_districts",
+ *     uniqueConstraints={
+ *         @ORM\UniqueConstraint(name="consular_district_code_unique", columns="code"),
+ *         @ORM\UniqueConstraint(name="consular_district_referent_tag_unique", columns="referent_tag_id")
+ *     }
+ * )
+ */
+class ConsularDistrict
```
Comment (ottaviano) [inline src/Entity/ConsularDistrict.php]: This entity should represent the table of consular districts (little different from existing `District` entity):
image
which has `id`, `countries`, `cities` and `code` columns.
Could you update this entity for matching the source of data?

File: src/Controller/EnMarche/EventManager/ConsularEventManagerController.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+use AppBundle\Repository\MunicipalEventRepository;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route(path="/espace-consulaire", name="app_consular_event_manager_")
+ *
+ * @Security("is_granted('ROLE_CONSULAR')")
+ */
+class ConsularEventManagerController extends AbstractEventManagerController
```
Comment (ottaviano) [inline src/Controller/EnMarche/EventManager/ConsularEventManagerController.php]: could you move the Event part in another PR pls?

File: src/Entity/ConsularDistrict.php (hunk 5 above)
Comment (Tiriel) [reply src/Entity/ConsularDistrict.php]: Isn't the "N° circo" column the number, and the code a different thing?
